app/src/main/java/org/simple/clinic/teleconsultlog/teleconsultrecord/screen/TeleconsultRecordEvent.kt

```diff
@@ -3,17 +3,26 @@ package org.simple.clinic.teleconsultlog.teleconsultrecord.screen
 import org.simple.clinic.teleconsultlog.teleconsultrecord.Answer
 import org.simple.clinic.teleconsultlog.teleconsultrecord.TeleconsultRecordWithPrescribedDrugs
 import org.simple.clinic.teleconsultlog.teleconsultrecord.TeleconsultationType
+import org.simple.clinic.widgets.UiEvent
 
-sealed class TeleconsultRecordEvent
+sealed class TeleconsultRecordEvent : UiEvent
 
-object BackClicked : TeleconsultRecordEvent()
+object BackClicked : TeleconsultRecordEvent() {
+  override val analyticsName: String = "Teleconsult Record:Back Clicked"
+}
 
-data class TeleconsultRecordWithPrescribedDrugsLoaded(val teleconsultRecordWithPrescribedDrugs: TeleconsultRecordWithPrescribedDrugs?) : TeleconsultRecordEvent()
+data class TeleconsultRecordWithPrescribedDrugsLoaded(val teleconsultRecordWithPrescribedDrugs: TeleconsultRecordWithPrescribedDrugs?) : TeleconsultRecordEvent() {
+  override val analyticsName: String = "Teleconsult Record:Record with Prescribed Drugs Loaded"
+}
 
-object TeleconsultRecordCreated : TeleconsultRecordEvent()
+object TeleconsultRecordCreated : TeleconsultRecordEvent() {
+  override val analyticsName: String = "Teleconsult Record:Record Created"
```

Comment:
We should not be overriding analytics names for events which are not UI events unless we specifically want to capture them.